Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -1206,6 +1206,31 @@ private extension PurchasesOrchestrator {
         }
     }
 
+    #if compiler(>=5.10) && !os(tvOS) && !os(watchOS) && !os(visionOS)
+
+    @available(iOS 16.4, macOS 14.4, *)
+    @available(tvOS, unavailable)
+    @available(watchOS, unavailable)
+    @available(visionOS, unavailable)
+    func trackApplePurchaseIntentReceivedIfNeeded(purchaseIntent: any StoreKit2PurchaseIntentType) {
+        let offerId: String?
+        let offerType: String?
+
+        if #available(iOS 18.0, macOS 15.0, *) {
```

Comment:
I see we're also doing `#if compiler(>=6.0)` below when accessing `purchaseIntent.offer`. Do we need to do the same here?

Reply:
I think probably so - [PurchaseIntent.offer](https://developer.apple.com/documentation/storekit/purchaseintent/offer) was introduced in iOS 18.0, which was released with Xcode 16 / Swift compiler 6.0.

@ajpallares I'd definitely recommend running all of the tests on this PR with `@RCGitBot please test` to make sure it still compiles on older Xcode versions. I had a pretty tough time when writing this code for the first time because of all of the device & OS compatibility requirements.

Reply:
@ajpallares In case you haven't seen it before, https://xcodereleases.com/ is the BEST resource out there for looking up what Swift versions shipped with what xcode & iOS versions. I don't know how I'd do my job without that website 😅

Reply:
OMG THANK YOU! I didn't know it but it goes directly into my favorites! 🤩